### PR TITLE
Add version requirement on async-http-client.

### PR DESCRIPTION
**Pull request by tachyonics**
*Issue #, if available:*

*Description of changes:* Add version requirement on async-http-client due to the use of `EventLoopGroup.singleton`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ jobs:
   run-codeql-linux:
     name: Run CodeQL on Linux
     runs-on: ubuntu-latest
+    container: swift:5.8
```
Comment (luiabrah) [inline .github/workflows/codeql.yml:13]: Should this be 5.9?

File: Package.swift [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:5.6
+// swift-tools-version:5.7
```
Comment (luiabrah) [inline Package.swift:1]: Should we increase to 5.9?

**Review by luiabrah — APPROVED** [review]

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (tachyonics) [reply .github/workflows/codeql.yml:13]: It doesn't look like Github currently supports CodeQL for 5.9.

File: Package.swift (hunk 2 above)
Comment (tachyonics) [reply Package.swift:1]: Our [support policy](https://github.com/amzn/smoke-aws/blob/main/docs/Support_Policy.md) states that we have a trailing window of supporting the two most recent compiler versions and we can drop support for the third most recent compiler version six months after the release of the latest compiler version. So according to this we should support 5.7 until the 18th of March 2024.

**Pull request merged**